### PR TITLE
Face recognition

### PR DESCRIPTION
**Pull request by druefena**
First integration of face recognition into Aana SDK. There are three separate deployments, namely face detection, face feature extraction, and face database, which can be connected to perform end-to-end face recognition.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -54,6 +54,8 @@ torchvision = ">=0.15.0"
 vllm = "^0.5.0.post1"
 yt-dlp = ">=2024.4.9"
 
+scikit-image = "^0.24.0"
```
Comment (evanderiel) [inline pyproject.toml:57]: Please keep dependencies alphabetized

File: aana/deployments/face_database_deployment.py [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+                distances, indices = self.index.search(query_features, 1)
+                results = []
+                for i, (distances_i, indices_i) in enumerate(
+                    zip(distances, indices, strict=False)
+                ):
+                    distance = distances_i[0]
+                    index = indices_i[0]
+                    face_norm = face_features["norms"][i][0]
+
+                    if index == -1:
```
[the remark this answers is not in the document]
Comment (evanderiel) [reply aana/deployments/face_database_deployment.py:220]: Please also consider moving the body of the for loops to separate functions

**Review by evanderiel — APPROVED** [review]
Looks good but some comments. Some more consistency in style would be good but I think ruff can take care of that

**Comment by appoose** [comment]
@druefena What is the status here on the changes @evanderiel  suggested?

**Comment by druefena** [comment]
I have not yet looked into these.

**Comment by druefena** [comment]
When I got back from vacation we decided to focus on the GSR task, and then three weeks ago there was a github commit error by a team member and I was told that the PR was gone. Did not even know it was back alive until now.

**Comment by appoose** [comment]
We can reopen the PR. Looks like it was closed in mistake. But we need to have a plan to finsh it off.

**Comment by appoose** [comment]
Reopening the PR since it is completed.

**Pull request closed without merging**